python/ray/autoscaler/_private/monitor.py

```diff
@@ -134,6 +134,7 @@ def __init__(
         self,
         address: str,
         autoscaling_config: Union[str, Callable[[], Dict[str, Any]]],
+        session_name: str,
```

Comment:
This will break things -- `Monitor()` is accidentally a public API.

What is `session_name` and how is it generated?

Reply:
And can Monitor fetch `session_name` from somewhere without having it passed in as an argument?

Reply:
Should we talk in person? I'd like to know exactly how this API is public?

Reply:
> (also it doesn't make sense that public APIs are broken by adding a new arg?)

If the argument is required, it will break things.

Reply:
> > And can Monitor fetch session_name from somewhere without having it passed in as an argument?
> 
> 
> 
> Hmm.. it is tricky I think. Maybe we can pass the env var via subprocess and get the value from it, but it seems pretty hacky. This also should work in the prod. 

The only semi-reasonable way to do it I can think of would be to store the info in the GCS.

Reply:
I think getting the data from internal KV is a viable solution if you think that's the best?